File: API/Entities/Consumable.cs
using System.Collections.Generic;

namespace API.Entities
{
  public class Consumable
  {
    public int Id { get; set; }
    public int SapId { get; set; }
    public string Description { get; set; }
    public string UnitOfMeasure { get; set; }
    public bool IsSite { get; set; }
    
    // the order quantity - this application doesn't take 
    public int Quantity { get; set; }
    
    public List<AreaOfWork> AreaOfWorks { get; set; } = new List<AreaOfWork>();
  }
}

File: client-app/src/app/models/consumable.ts
import { AreaOfWork } from './areaOfWork';

export interface Consumable {
  id: number;
  sapId: number;
  description: string;
  unitOfMeasure: string;
  isPrd: boolean;

  areaOfWorks: AreaOfWork[];

  orderQuantity?: number;
}

export interface ConsumableSubmitValues {
  sapId: number;
  description: string;
  unitOfMeasure: string;
  isPrd: boolean;

  orderQuantity?: number;
}

export class ConsumableFormValues {
  sapId: number | undefined = undefined;
  description: string = '';
  unitOfMeasure: string = '';
  isPrd: boolean = false;

  ServiceOrderIds: number[] = [];

  constructor(consumable?: ConsumableFormValues) {
    if (consumable) {
      this.sapId = consumable.sapId;
      this.description = consumable.description;
      this.unitOfMeasure = consumable.unitOfMeasure;
      this.isPrd = consumable.isPrd;
      this.ServiceOrderIds = consumable.ServiceOrderIds;
    }
  }
}
